### PR TITLE
feat(memcache): generate library

### PR DESCRIPTION
**Pull request by devjgm**
Fixes: #8152

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8153)

### CODE REVIEW

File: google/cloud/memcache/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+* Bazel and CMake builds
+
+## Documentation
+
+* Official documentation about the [Cloud Memorystore for Memcached API][cloud-service-docs] service
+* [Reference doxygen documentation][doxygen-link] for each release of this
+  client library
+* Detailed header comments in our [public `.h`][source-link] files
+
+[cloud-service-docs]: https://cloud.google.com/memcache
```
Comment (coryan) [inline google/cloud/memcache/README.md:29]: This link is broken

File: google/cloud/memcache/doc/main.dox [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+For operations that do not return a value the library simply returns
+`google::cloud::Status`.
+
+### Retry, Backoff, and Idempotency Policies.
+
+The library automatically retries requests that fail with transient errors, and
+uses [exponential backoff] to backoff between retries. Application developers
+can override the default policies.
+
+[cloud-service-docs]: https://cloud.google.com/memcache
```
Comment (coryan) [inline google/cloud/memcache/doc/main.dox:85]: Likewise, broken link

**Review by coryan — CHANGES_REQUESTED** [review]

File: google/cloud/memcache/README.md (hunk 1 above)
Comment (devjgm) [reply google/cloud/memcache/README.md]: Oops. Fixed.

File: google/cloud/memcache/doc/main.dox (hunk 2 above)
Comment (devjgm) [reply google/cloud/memcache/doc/main.dox]: Fixed.

**Review by coryan — APPROVED** [review]

**Review by coryan — APPROVED** [review]

**Pull request merged**